### DISCUSSION TITLE
Touch pattern recognition

### DISCUSSION THREAD

**Discussion by sfxsfl** [Bodmer/TFT_eSPI#2012] [Q&A - Touch]
Hi and good evening,
first of all I would like to thank you for this great masterpiece. I've started with TFT_eSPI already a while ago on ILI9341 with
ESP8266 and it works out of the box with TFT_SPI_OVERLAP.

I would like to extend my Touch screen pattern recognition in such way to recognize different states like, Button pressed (not longer than 1sec), Button long pressed (longer than 2sec) and maybe some sliding across the screen, e.g. left to right and right to left. 

In my application I have buttons and sliders. The challenge for me is to handle the different behavior of the screen elements.
A slider needs to recognize slight changes on the screen (cursor movements) and not just the pressed and unpressed state. 

I'm struggling recognizing the patterns in the loop() with getTouch(&xPos, &yPos). Currently, I'm sampling the data 
with roughly 50ms. This seems to work for the slider. 

Any hint or idea is highly appreciated.  

Thanks in advance
Stefan

**Comment by Bodmer** [comment]
The TFT_eWidget may fit your needs and is available via the Arduino IDE library manager.

**Reply by sfxsfl** [reply]
Thanks a lot for the quick feedback. I will have a look and let you know ... BR Stefan

**Reply by Bodmer** [reply]
Info here may help too #1438
